### PR TITLE
TD-Balance-Summer-2020-#2

### PR DESCRIPTION
**Pull request by Orb370**
So I have not been idle since #18273. These are more mild changes. I was originally focusing on air changes, but most of those turned out to be unwarranted, so this is more of an AA/QoL PR. These were tested under the "Fresh Air" label for balance maps.

I'm mostly satisfied with these changes, but I would like to do one more week of testing. Just putting it here for discussion at the moment.

**GDI AA**

GDI mobile AA (APC) has been really weak in testing so I gave it a few buffs.

`APC: AA Range: 6-> 7, Vision: 6 -> 7, Custom build time removed`

Reverting some of the changes made by AoA in his patch #15150, as they impacted the APC's AA too much. Also, APC AG was the issue, not AA, which the other changes did fix.

`APC: Prioritize Air`

It's very easy to distract APCs with ground targets and then destroy them with Orcas. Aircraft are fast and difficult to right click for an attack. This change means APCs will prioritize aircraft when searching for a new target (stop command, current target is killed).

**NOD AA**

Nod's AA on the other hand is too strong and makes using air against them an unwise choice, particularly in the late game.

`Stealth Tank: Weapon split, AA weapon: Minimum range 0.5 -> 2.5`

Invisible AA isn't great. As the aircraft user you have 0 counter play.  It makes aircraft pointless late game vs Nod since there could be AA anywhere on the map without you knowing. This change forces stealth tanks to give some distance before firing, giving aircraft a split second to dodge the missiles. It makes it more difficult for stealth tanks to hit moving aircraft.

This decision is also backed up by the stealth tank model, since it doesn't really seem like it can shoot straight up. [Image](https://gamepedia.cursecdn.com/cnc_gamepedia_en/e/e7/TD_Stealth_Tank_DOS_Manual.gif?version=b0932e3eb1e1ca4a6d201ae1e73028de)

`Sam Site: Construction animation lengthened (Tick 30 -> 50)`

The Sam Site constructs much faster than every other defense in TD. It can be placed to kill aircraft with no real response time. The construction time now closely matches the AGT.

**Balance Changes**

`Flame Tank: Vision 5 -> 6`

Flame tanks feel clunky to use. This puts their vision in line with other tanks (MT/LT's have 6 vision). This should make it easier to avoid defenders and find weak spots.

`MRLS (Rocket Launcher): Light 100 -> 90`

The MRLS is a little too potent vs light vehicles, and can wipe out several bikes in one volley. This damage change still allows it to one volley any light vehicle, but makes chip/AOE damage less punishing.

**QoL/Polish**

`Harvester: Vision 4 -> 5, Loss Notification added`

So I'll start with the major change. I've added a loss notification for the harvester (spliced together by Kato). I've really liked it in testing and I've only heard positive comments from players (I asked a new player about it and he said he didn't even notice it, so it's not out of place).

Harvester counts in TD are really important, and this has had a positive impact on gameplay. For one, if you don't notice a harassment and lose 4 harvesters, you just lose. With this change a successful harassment gives you an edge, not an instant win. (Note: In TD just knowing you lost a harvester doesn't mean you'll save the rest either, this just gives you a chance to respond).

The change also helps you know when to rebuild eco, or even just how many harvesters you lost in the first place.

Vision change: Makes it slightly easier to notice/defend harvesters (for reference bikes have 6 weapon range).

`Mammoth: Missile range 5 -> 4.75`
`Orca: Range 5 -> 4.75`

Mammoth missiles were 5 range while the cannons were 4.75. This means if you attack a structure the mammoth will only get in range of the missiles and will be unable to fire the cannons. I've reduced the missile range to avoid this. 

However, what this did lead to was Orcas being able to fire on Mammoths without them being able to fire back. I've adjusted the Orca missile range down so they're the same range again.

`Helipad: Description: Produces and repairs helicopters`
`AGT: Description: All-purpose defensive structure.\n  Strong vs Aircraft, Infantry\n  Weak vs Tanks`

Old helipad description said it rearmed aircraft (which is not the case), so I've changed it. 

AGT's are a generalist tower, but are actually weaker to tanks than infantry, so I've switched those in the description.

### CODE REVIEW

**Comment by Orb370** [comment]
Based on testing today I think I'm going to remove the Harvester vision buff, so no need to comment on that (unless you think it's a good change).

**Comment by Orb370** [comment]
*tumbleweed*

Well, I guess we're all good to go. No further issues came up in testing.

**Comment by JMegaTank** [comment]
Well done Orb your doing a great job so far on the balance in TD

File: mods/cnc/rules/vehicles.yaml [hunk 1]
```diff
@@ -76,6 +76,8 @@ HARV:
 		Type: Heavy
 	RevealsShroud:
 		Range: 4c0
+    ActorLostNotification:
```
Comment (abcdefg30) [inline mods/cnc/rules/vehicles.yaml]: This should use a tab.

Comment (Orb370) [reply mods/cnc/rules/vehicles.yaml]: Fixed

**Review by abcdefg30 — APPROVED** [review]

**Review by Smittytron — APPROVED** [review]

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/646d8fc627299cb70c5664bfb49413daeb2f85ce

**Pull request merged**